store/src/main/scala/geotrellis/store/GeoTrellisRasterSource.scala

```diff
@@ -139,25 +139,12 @@ object GeoTrellisRasterSource {
   def getClosestResolution[T](
     grids: Seq[T],
     cellSize: CellSize,
-    strategy: OverviewStrategy = AutoHigherResolution
-  )(implicit f: T => CellSize): Option[T] = {
-    val maxResultion = Some(grids.minBy(g => f(g).resolution))
-
-    strategy match {
-      case AutoHigherResolution =>
-        grids // overviews can have erased extent information
-          .map { v => (cellSize.resolution - f(v).resolution) -> v }
-          .filter(_._1 >= 0)
-          .sortBy(_._1)
-          .map(_._2)
-          .headOption
-          .orElse(maxResultion)
-      case Auto(n) =>
-        val sorted = grids.sortBy(v => math.abs(cellSize.resolution - f(v).resolution))
-        sorted.lift(n).orElse(sorted.lastOption) // n can be out of bounds,
-      // makes only overview lookup as overview position is important
-      case Base => maxResultion
-    }
+    strategy: OverviewStrategy = OverviewStrategy.DEFAULT
+  )(implicit f: T => CellSize): T = {
```

Comment:
Ah, should have mentioned this sooner. 

It was Option[Int] because some operations could return an out of bounds index.  It's not clear how GDAL handles this situation, so we decided that it would be simpler if OverviewStrategy.selectOverview always returned an in bounds index. Which it now does due to additional error handling there. Before this change, we were doing an `option.get` everywhere which just pushes the throw further down. In that case, I wasn't sure how we would even handle getting a bad overview. 

I can add some additional docstrings to note that selectOverview should always return a clamped index, this was apparently the intent of the code based [on a test](https://github.com/locationtech/geotrellis/pull/3200/files#diff-28fce00d2a6d9757847488e95fe46c87R194) that predates me picking this task up.

Reply:
When it can’t fine an overview the idea was to use the Base layer, since it looks like what GDAL does.

Reply:
How does that reconcile with the idea that we've decided to include the base layer in the resolutions list in all RasterSource implementations?

Reply:
Perhaps this discussion should be moved offline as well?

Reply:
After a discussion we decided on the following:
- Revert OverviewStrategy.DEFAULT = AutoHigherResolution to make the default consistent with the old behavior for now. We'll address changing the default at a later date
- Update AutoHigherResolution implementation to match the docstring and old behavior, and verify it with tests
- OverviewStrategy.selectOverview still clamps to available indices. Each overview strategy can perform custom clamping behavior, like AutoHigher above, but in the absence of that, the index will be clamped to the nearest valid value, e.g. -1 -> 0 or list.size + 5 -> list.size - 1
- Revert tests where possible to use the AutoHigher default in order to appropriately test the old behavior.